### ISSUE TITLE
Invalid generation for maps of integers

### ISSUE THREAD

**Issue by mt-inside** [kube-rs/kopium#45]
I'm quite new to rust so forgive me if this is stupid.

I've generated a file for Istio's [DestinationRule](https://istio.io/latest/docs/reference/config/networking/destination-rule/) type: `kopium destinationrules.networking.istio.io --docs > destinationrule.rs`

The CRD has various fields typed as `integer` in different ways (I'm also not an OpenAPI expert).
Mostly these work fine:

```
                        port:                                                                 
                          properties:                                                         
                            number:                                                           
                              type: integer   
```
becomes
```
pub number: Option<i64>,
```

Something like
```
                            consecutiveErrors:                                                
                              format: int32                                                   
                              type: integer 
```
becomes
```
    pub consecutiveErrors: Option<i32>,
```

The problem _seems_ to be objects with `additionalProperties` eg
```
                                      to:                                                     
                                        additionalProperties:                                 
                                          type: integer                                       
                                        description: Map of upstream localities to traffic distribution weights.                       
                                        type: object   
```
which becomes
```
    pub to: Option<BTreeMap<String, Integer>>,
```

This doesn't compile because the `Integer` type is unknown.
I assume this is really meant to be `i64` (or `i32` if a `format` is given).
I did try a couple of things under the assumption it is really meant to be `Integer` and we're just missing a `use`, but serde_json doesn't have an `Integer`, and it's not `num::Integer` because that's a trait and they're not used like trait objects.

**Comment by clux** [comment]
Ah, thanks a lot. I was a bit unsure of the `additionalProperties` stuff when implementing that.

Do you have an example instance of a `destinationrule` that has consecutive errors field set? It would be helpful to start this off with some failing tests :-)

**Comment by clux** [comment]
Partial fix for istio destinationrules have been applied which at least fixes integer generation for maps which this issue was originally about.

Unfortunately, the crd is still blocked by another issue (#31) because it has an arbitrary status object which we don't currently detect.

**Comment by clux** [comment]
Anyway, thanks for the bug. If you see anything else weird, don't hesitate to file more.

**Comment by mt-inside** [comment]
Hey, sorry got dragged off that project, only just seen your responses. Thanks so much for fixing this!

Yeah I realise there's more issues with DestinationRule like the empty Status, I'm working round those atm with manual modification of the generated code. Appreciate the work you're putting in on this project!
